common/kubernetes-membership-scheme/src/main/java/org/wso2/carbon/membership/scheme/kubernetes/api/KubernetesHttpsApiEndpoint.java

```diff
@@ -92,4 +96,8 @@ public boolean verify(String hostname, SSLSession session) {
             HttpsURLConnection.setDefaultHostnameVerifier(dummyHostVerifier);
         } catch (Exception ignored) {}
     }
+
+    private String getServiceAccountToken() throws IOException {
+        return new String(Files.readAllBytes(Paths.get(KubernetesMembershipSchemeConstants.BEARER_TOKEN_FILE_LOCATION)));
```

Comment:
A file containing an API token for a pod’s service account is automatically mounted at /var/run/secrets/kubernetes.io/serviceaccount/token. That token can be used to make API calls as the pod’s service account.

Token file persisits until the rolling update being performed in container.

[1] https://access.redhat.com/documentation/en/openshift-enterprise/3.1/developer-guide/chapter-4-service-accounts

[2]  
https://github.com/pires/hazelcast-kubernetes-bootstrapper/blob/master/src/main/java/com/github/pires/hazelcast/HazelcastDiscoveryController.java#L1